src/authenticate.rs

```diff
@@ -4,8 +4,29 @@ use datatype::{AccessToken, Error, Url};
 use http::{Client, Response};
 
 
+#[derive(RustcEncodable)]
+#[allow(non_snake_case)]
+struct RegistrationPayload {
+    deviceId: String,
+    ttl: u32
+}
+
+/// Register with the specified auth gateway server to retrieve a new pkcs#12 bundle.
+pub fn pkcs12(server: Url, device_id: String, ttl: u32, client: &Client) -> Result<Vec<u8>, Error> {
+    debug!("registering at {}", server);
+    let body = try!(json::encode(&RegistrationPayload { deviceId: device_id, ttl: ttl }));
+    let resp_rx = client.post(server, Some(body.into_bytes()));
+    let resp    = resp_rx.recv().expect("no authenticate response received");
```

Comment:
yes, it is a critical error if the transmitter closes the channel before the receiver can process it